### PR TITLE
more consistent MSVC++ project settings 

### PR DESCRIPTION
**Pull request by lucaspimentel**

### CODE REVIEW

File: src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj [hunk 1]
```diff
@@ -113,37 +114,40 @@
...
       <Message>Remove potentially locked outputs (dll/pdb/xml)</Message>
     </PreBuildEvent>
-    <PostBuildEvent>
-      <Command>IF EXIST $(ProjectDir)\post-build-events-cpp.bat CALL $(ProjectDir)\post-build-events-cpp.bat "$(SolutionDir)"</Command>
-    </PostBuildEvent>
-    <PostBuildEvent>
-      <Message>Copy profiler for other projects</Message>
-    </PostBuildEvent>
+    <PostBuildEvent />
+    <PostBuildEvent />
```
Comment (colin-higgins) [inline src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj]: Can probably delete these two lines.

File: src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj [hunk 2]
```diff
@@ -170,6 +181,10 @@
       <OptimizeReferences>true</OptimizeReferences>
       <ModuleDefinitionFile>.\Datadog.Trace.ClrProfiler.Native.def</ModuleDefinitionFile>
     </Link>
+    <PreBuildEvent>
+      <Command>IF EXIST $(SolutionDir)\pre-build-events-cpp.bat CALL $(SolutionDir)\pre-build-events-cpp.bat  "$(TargetPath)"  "$(TargetFileName)"  "$(TargetDir)"  "$(TargetName)"</Command>
```
Comment (colin-higgins) [inline src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj]: Is this called twice?

Comment (zacharycmontoya) [reply src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj]: No, it looks like it's getting called once per Debug/Release | Win32/x64 build.

File: src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.vcxproj [hunk 3]
```diff
@@ -61,84 +61,86 @@
...
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
     <ClCompile>
       <WarningLevel>Level3</WarningLevel>
       <Optimization>Disabled</Optimization>
-      <TreatWarningAsError>false</TreatWarningAsError>
       <RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>
       <LanguageStandard>stdcpp17</LanguageStandard>
-      <PreprocessorDefinitions>_CRT_SECURE_NO_WARNINGS;_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <MultiProcessorCompilation>true</MultiProcessorCompilation>
+      <MultiProcessorCompilation>
```
Comment (zacharycmontoya) [inline src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.vcxproj]: Can you get rid of this in each of the ItemDefinitionGroup's?

File: test/Datadog.Trace.ClrProfiler.Native.Tests/Datadog.Trace.ClrProfiler.Native.Tests.vcxproj [hunk 4]
```diff
@@ -92,15 +98,14 @@
...
-      <PrecompiledHeader>Use</PrecompiledHeader>
-      <PrecompiledHeaderFile>pch.h</PrecompiledHeaderFile>
       <Optimization>Disabled</Optimization>
-      <PreprocessorDefinitions>WIN32;_DEBUG;_CONSOLE;_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING;_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <MinimalRebuild>true</MinimalRebuild>
       <BasicRuntimeChecks>EnableFastChecks</BasicRuntimeChecks>
       <RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>
-      <WarningLevel>Level3</WarningLevel>
       <LanguageStandard>stdcpp17</LanguageStandard>
+      <ConformanceMode>true</ConformanceMode>
```
Comment (zacharycmontoya) [inline test/Datadog.Trace.ClrProfiler.Native.Tests/Datadog.Trace.ClrProfiler.Native.Tests.vcxproj]: Debug|x64 has `<OmitFramePointers>false</OmitFramePointers>`. Did you mean to exclude that here?

File: src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj [hunk 5]
```diff
@@ -113,37 +114,40 @@
...
   </ItemDefinitionGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
     <ClCompile>
       <WarningLevel>Level3</WarningLevel>
       <Optimization>Disabled</Optimization>
-      <PreprocessorDefinitions>WIN32;_DEBUG;_WINDOWS;_USRDLL;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <SDLCheck>true</SDLCheck>
-      <ConformanceMode>true</ConformanceMode>
       <RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>
+      <ConformanceMode>true</ConformanceMode>
```
Comment (zacharycmontoya) [inline src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj]: Debug|x64 has `<OmitFramePointers>false</OmitFramePointers>` which is missing here. Did you mean to exclude it here?

Comment (lucaspimentel) [reply src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj]: Omitting frame pointers is an optimization that can make debugging harder, so they are omitted (`true`) in Release builds and included (`false`) in Debug builds.
https://docs.microsoft.com/en-us/cpp/build/reference/oy-frame-pointer-omission

Comment (lucaspimentel) [reply src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj]: But `In x64 compilers, /Oy and /Oy- are not available`, which might explain the discrepancy between `x86` and `x64`.

File: src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.vcxproj (hunk 3 above)
Comment (lucaspimentel) [reply src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.vcxproj]: Sure. To be honest, I didn't look at any of the files, I just edited the setting in the UI, where you can choose multiple projects, configurations, and platforms and see if they have different settings.

Comment (lucaspimentel) [reply src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.vcxproj]: Actually, I'm going to time build with and without this and leave the fastest options, which is probable `true`.

File: src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj (hunk 2 above)
Comment (lucaspimentel) [reply src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj]: Right. It was set for some configurations but not others.

File: src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.vcxproj (hunk 3 above)
Comment (lucaspimentel) [reply src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.vcxproj]: image

image

Total time to rebuild all 3 C++ projects went down to about half by turning this on. Note that it required turning "create precompiled header off" in the Tests project.

8fde4e4f52416005b6ffcf295422decc83b1487c

File: test/Datadog.Trace.ClrProfiler.Native.Tests/Datadog.Trace.ClrProfiler.Native.Tests.vcxproj (hunk 4 above)
Comment (lucaspimentel) [reply test/Datadog.Trace.ClrProfiler.Native.Tests/Datadog.Trace.ClrProfiler.Native.Tests.vcxproj]: See https://github.com/DataDog/dd-trace-dotnet/pull/387#discussion_r292648935

File: src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj (hunk 1 above)
Comment (lucaspimentel) [reply src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj]: 17d28deea74cb39eb881e2802ded8f545708f0b6

**Pull request merged**